pxr/usd/usdUtils/dependencies.cpp

```diff
@@ -655,8 +655,28 @@ class _AssetLocalizer {
             auto &fileAnalyzer = destFilePathAndAnalyzer.second;
 
             if (!fileAnalyzer.GetLayer()) {
-                _fileCopyMap.emplace_back(fileAnalyzer.GetFilePath(),
-                                          destFilePath);
+                const std::string srcFilePath = fileAnalyzer.GetFilePath();
+                if (TfStringContains(srcFilePath, "<UDIM>")) {
+                    // Special case for UDIMs to glob for UDIM numbers
+                    const std::string::size_type udimPos = 
+                        srcFilePath.find("<UDIM>");
+                    const std::string globPath = TfStringReplace(
+                        srcFilePath, "<UDIM>", "*");
+
+                    const std::vector<std::string> udimPaths = TfGlob(globPath);
```

Comment:
...and, having said above I wasn't concerned about this simpler logic, I now realize that we need to do the **same** kind of search that e.g. [_ResolveAssetAttribute](https://github.com/PixarAnimationStudios/USD/blob/dev/pxr/usdImaging/usdImaging/materialParamUtils.cpp#L156) does, because `TfGlob()` will only work, even on filesystems, for anchored or absolute paths.  So we need to find **one** actual UDIM file before we can run `TfGlob()`

 That's not a trivial amount of code, and I really don't advocate duplicating it.  I'm not sure UsdUtils is the _ideal_ place for introducing UDIM-support-logic, but it's not horrible, and it is at least accessible to UsdImaging.  So I would advocate, for now, refactoring that logic out of materialParamUtils.cpp to a public helper here.  If there are folks at Pixar who think we can find a better long-term home for it, we'll take on any further refactoring.